src/GovUk.Education.ExploreEducationStatistics.Publisher.Tests/Services/NotificationsServiceTests.cs

```diff
@@ -0,0 +1,222 @@
+                        Created = DateTime.UtcNow,
+                        NotifySubscribers = true,
+                        NotifiedOn = DateTime.UtcNow,
+                    },
+                },
+            };
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddRangeAsync(release);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var storageQueueService = new Mock<IStorageQueueService>(MockBehavior.Strict);
+            storageQueueService.Setup(mock => mock.AddMessages(
+                PublicationQueue,
+                It.IsAny<List<PublicationNotificationMessage>>()))
+                .Returns(Task.CompletedTask);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var notificationsService = BuildNotificationsService(
+                    contentDbContext, storageQueueService.Object);
+
+                await notificationsService.NotifySubscribersIfApplicable(release.Id);
+
+                storageQueueService.Verify(mock => mock.AddMessages(
+                    PublicationQueue,
+                It.IsAny<List<PublicationNotificationMessage>>()), Times.Never);
+            }
+        }
+
+        [Fact]
+        public async Task NotifySubscribersIfApplicable_None()
+        {
+            var storageQueueService = new Mock<IStorageQueueService>(MockBehavior.Strict);
+            storageQueueService.Setup(mock => mock.AddMessages(
+                PublicationQueue,
+                It.Is<List<PublicationNotificationMessage>>(list =>
+                    list.SequenceEqual(new List<PublicationNotificationMessage>()
+                    ))))
+                .Returns(Task.CompletedTask);
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var notificationsService = BuildNotificationsService(
+                    contentDbContext, storageQueueService.Object);
+
+                await notificationsService.NotifySubscribersIfApplicable();
+
+                storageQueueService.Verify(mock => mock.AddMessages(
+                    PublicationQueue,
+                It.Is<List<PublicationNotificationMessage>>(list =>
+                    list.SequenceEqual(new List<PublicationNotificationMessage>())
+                    )), Times.Never);
+            }
+        }
+
+        private static NotificationsService BuildNotificationsService(
+            ContentDbContext contentDbContext,
+            IStorageQueueService storageQueueService = null)
+        {
+            return new (
+                contentDbContext,
+                storageQueueService ?? Mock.Of<IStorageQueueService>());
```

Comment:
Add Strict here to the default mock and then the two tests `NotifyOnSet` and `None` can make use of it without needing to set up or verify anything since they both expect no interactions.